### PR TITLE
CM-104 Modificar users-be para que opere segun los permisos del usuar…

### PR DESCRIPTION
**Pull request by merlindk**
…io logueado

### CODE REVIEW

**Comment by FranciscodeMaussion** [comment]
test this please

**Comment by merlindk** [comment]
test this please

**Comment by merlindk** [comment]
test this please

File: src/main/java/com/cema/users/config/SecurityConfig.java [hunk 1]
```diff
@@ -75,16 +87,33 @@ public FilterRegistrationBean simpleCorsFilter() {
...
+                .authorizeRequests()
+                .expressionHandler(webExpressionHandler())
+                .antMatchers("/v1/authenticate", "/v2/api-docs",
+                        "/configuration/ui",
+                        "/swagger-resources/**",
+                        "/configuration/security",
+                        "/swagger-ui.html",
+                        "/webjars/**")
+                .permitAll()
+//                    .antMatchers("/v1/users/**")
```
Comment (FranciscodeMaussion) [inline src/main/java/com/cema/users/config/SecurityConfig.java]: comments

File: src/main/java/com/cema/users/controllers/UserController.java [hunk 2]
```diff
@@ -73,11 +73,15 @@ public Controller(CemaUserRepository cemaUserRepository, UserMapping userMapping
         if (cemaUser == null) {
             throw new UserNotFoundException(String.format(Messages.USER_DOES_NOT_EXISTS, userName));
         }
+        if (!authorizationService.isOnTheSameEstablishment(cemaUser.getEstablishmentCuig())) {
+            throw new UnauthorizedException(String.format(Messages.OUTSIDE_ESTABLISHMENT, cemaUser.getEstablishmentCuig()));
```
Comment (FranciscodeMaussion) [inline src/main/java/com/cema/users/controllers/UserController.java:78]: maybe an @ApiResponse should be added?

File: src/main/java/com/cema/users/controllers/UserController.java [hunk 3]
```diff
@@ -96,19 +100,26 @@ public Controller(CemaUserRepository cemaUserRepository, UserMapping userMapping
         user.setUserName(userName);
         LOG.info("Request to register user: {}", userName);
 
+        if (!authorizationService.isOnTheSameEstablishment(user.getEstablishmentCuig())) {
+            throw new UnauthorizedException(String.format(Messages.OUTSIDE_ESTABLISHMENT, user.getEstablishmentCuig()));
```
Comment (FranciscodeMaussion) [inline src/main/java/com/cema/users/controllers/UserController.java:106]: maybe an @ApiResponse should be added?

File: src/main/java/com/cema/users/controllers/UserController.java [hunk 4]
```diff
@@ -96,19 +100,26 @@ public Controller(CemaUserRepository cemaUserRepository, UserMapping userMapping
...
+        }
+        if(!authorizationService.isAdmin() && user.getRole().equalsIgnoreCase(Roles.ADMIN)){
+            throw new UnauthorizedException(String.format(Messages.ACTION_NOT_ALLOWED, user.getRole()));
```
Comment (FranciscodeMaussion) [inline src/main/java/com/cema/users/controllers/UserController.java:109]: maybe an @ApiResponse should be added?

File: src/main/java/com/cema/users/controllers/UserController.java [hunk 5]
```diff
@@ -125,6 +136,9 @@ public Controller(CemaUserRepository cemaUserRepository, UserMapping userMapping
         userName = userName.toLowerCase();
         CemaUser cemaUser = cemaUserRepository.findCemaUserByUserName(userName);
         if (cemaUser != null) {
+            if (!authorizationService.isOnTheSameEstablishment(cemaUser.getEstablishmentCuig())) {
+                throw new UnauthorizedException(String.format(Messages.OUTSIDE_ESTABLISHMENT, cemaUser.getEstablishmentCuig()));
```
Comment (FranciscodeMaussion) [inline src/main/java/com/cema/users/controllers/UserController.java:143]: maybe an @ApiResponse should be added?

File: src/test/java/com/cema/users/controllers/UserControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+    }
+
+    @Test
+    public void registerShouldAlwaysReturnCreatedWhenUserAddedCorrectly() {
+        CemaUser cemaUser = new CemaUser();
+        User user = new User();
+        String userName = "merlin";
+        user.setUserName(userName);
+        user.setEstablishmentCuig(cuig);
+        user.setRole("PATRON");
```
Comment (FranciscodeMaussion) [inline src/test/java/com/cema/users/controllers/UserControllerTest.java]: we could save roles in a constant

**Review by FranciscodeMaussion — APPROVED** [review]

**Review by FranciscodeMaussion — APPROVED** [review]

**Comment by FranciscodeMaussion** [comment]
test this please

**Comment by FranciscodeMaussion** [comment]
test this please

**Pull request merged**